python/ml4ir/base/model/optimizer.py

```diff
@@ -5,58 +5,61 @@
 
 
 def get_optimizer(
-    optimizer_key: str,
-    learning_rate: float,
+    optimizer_key: str = "SGD",
+    with_exp_decay: bool = False,
+    initial_learning_rate: float = 0.01,
     learning_rate_decay: float = 1.0,
     learning_rate_decay_steps: int = 1000000,
-    gradient_clip_value: float = 1000000,
+    **kwargs
 ) -> tf_optimizers.Optimizer:
     """
     This function defines the optimizer used by ml4ir.
-    Users have the option to define an ExponentialDecay learning rate schedule
+    By default, if no arguments are passed the function returns
+    vanilla SGD. Given a valid optimizer key, the function will return
+    the passed optimizer, with its default params. You may pass any other
+    valid arguments to the function to control aspects of the optimizer,
+    according to the official Tensorflow documentation. See References below.
+
+    Users may use an ExponentialDecay learning rate schedule.
 
     Arguments:
-        optimizer_key: string optimizer name to be used as defined under ml4ir.base.config.keys.OptimizerKey
-        learning_rate: floating point learning rate for the optimizer
-        learning_rate_decay: floating point rate at which the learning rate will be decayed every learning_rate_decay_steps
-        learning_rate_decay_steps: int representing number of iterations after which learning rate will be decreased exponentially
-        gradient_clip_value: float value representing the clipvalue for gradient updates. Not setting this to a reasonable value based on the model will lead to gradient explosion and NaN losses.
+        optimizer_key: string optimizer name to be used as defined
+         under tensorflow.keras.optimizers
+        with_exp_decay: Set to True, to add Exponential Decay schedule
+        learning_rate_decay,learning_rate_decay, learning_rate_decay_steps:
+        Params controlling the decay schedule,
+         cf. https://www.tensorflow.org/api_docs/python/tf/keras/optimizers/schedules/ExponentialDecay
+
+    You may pass any other valid arguments to the function to control aspects of the optimizer,
+    according to the official tensforflow documentation.
 
     References:
         https://www.tensorflow.org/api_docs/python/tf/keras/optimizers/Optimizer
         https://www.tensorflow.org/api_docs/python/tf/keras/optimizers/schedules/ExponentialDecay
 
+
+    Examples:
+        get_optimizer() -> SGD optimizer
+        get_optimizer('Adam', learning_rate=0.002) -> Adam, with lr=0.002
     FIXME:
         Define all arguments overriding tensorflow defaults in a separate file
         for visibility with ml4ir users
+
     """
     # Define an exponential learning rate decay schedule
     learning_rate_schedule = ExponentialDecay(
```

Comment:
I think that schedulers should in their own `get_scheduler` function. But good for now.